### DISCUSSION TITLE
Why isn't `Engine` dispose implemented?

### DISCUSSION THREAD

**Discussion by scarletquasar** [sebastienros/jint#1387] [Q&A]
As I see, it is no-op for now. Are you people accepting PR's? I really do not want to make a fork

**Comment by Genteure** [comment]
I'm not a maintainer but yes this project accept PRs.

quote from [here](https://github.com/sebastienros/jint/issues/1368#issuecomment-1329449422):

> Well first of all, the Dispose is a lie, in a sense 😉 Engine currently doesn't currently do anything in Dispose, but it's a point for things to happen and it's a good idea to account for it. It's part of getting the API "right", hopefully. We are iterating here.

It doesn't do anything because currently it doesn't need to do anything, but `Engine` still implement `IDisposable` because it might be needed later.

Discussion about making `Engine` implement `IDisposable`: #423 

While on the topic of API, as @lahma expressed in various issues and PRs, we'd like to make the public API small and clean so when Jint 3.0.0 stable eventually come, the implementation can still be changed without breaking public API.
That means:
1. try to hide as much as possible (making them `internal` instead of `public`), if people are using it then we make them public again and add a test in `Jint.Tests.PublicInterface`.
2. make the public API able to accommodate upcoming implementation changes, for example, when something added in `Engine` that _do_ need disposing.

I'm sure everything is up for discussion, if you want to remove `IDispose`, what's your reasoning? or if you want to implement `Dispose`, what do you want to call?

**Reply by lahma** [reply]
This is a great answer and I think it covers the current situation, nothing identified yet to dispose.

**Reply by scarletquasar** [reply]
I would need it in order to make an "ephemeral" engine, currenly working with multithreading in my runtime and making workarounds to keep the performance in focus. 

I wanted to clear all the "ephemeral" engine content and active operations (like promises). Going to study more about that and possibly make a pull request soon.
